test/colorado_lottery_test.rb

```diff
@@ -0,0 +1,222 @@
+    @lottery.register_contestant(@winston, @mega_millions)
+    expected = {
+                "Pick 4" => [@alexander],
+                "Mega Millions" => [@alexander, @frederick, @winston],
+                "Cash 5" => [@winston]
+                }
+    assert_equal expected, @lottery.registered_contestants
+
+    @lottery.register_contestant(@grace, @mega_millions)
+    @lottery.register_contestant(@grace, @cash_5)
+    @lottery.register_contestant(@grace, @pick_4)
+    expected = {
+                "Pick 4" => [@alexander, @grace],
+                "Mega Millions" => [@alexander, @frederick, @winston, @grace],
+                "Cash 5" => [@winston, @grace]
+                }
+    assert_equal expected, @lottery.registered_contestants
+  end
+
+  def test_it_returns_eligible_contestants
+    @lottery.register_contestant(@alexander, @pick_4)
+    @lottery.register_contestant(@alexander, @mega_millions)
+    @lottery.register_contestant(@frederick, @mega_millions)
+    @lottery.register_contestant(@winston, @cash_5)
+    @lottery.register_contestant(@winston, @mega_millions)
+    @lottery.register_contestant(@grace, @mega_millions)
+    @lottery.register_contestant(@grace, @cash_5)
+    @lottery.register_contestant(@grace, @pick_4)
+
+    assert_equal [@alexander, @grace],@lottery.eligible_contestants(@pick_4)
+    assert_equal [@winston, @grace], @lottery.eligible_contestants(@cash_5)
+    assert_equal [@alexander, @frederick, @winston, @grace], @lottery.eligible_contestants(@mega_millions)
+  end
+
+  def test_it_charges_and_returns_current_contestants
+    @lottery.register_contestant(@alexander, @pick_4)
+    @lottery.register_contestant(@alexander, @mega_millions)
+    @lottery.register_contestant(@frederick, @mega_millions)
+    @lottery.register_contestant(@winston, @cash_5)
+    @lottery.register_contestant(@winston, @mega_millions)
+    @lottery.register_contestant(@grace, @mega_millions)
+    @lottery.register_contestant(@grace, @cash_5)
+    @lottery.register_contestant(@grace, @pick_4)
+
+
+    @lottery.charge_contestants(@cash_5)
+    expected = {@cash_5 => ["Winston Churchill", "Grace Hopper"]}
+    assert_equal expected, @lottery.current_contestants
+    assert_equal 19, @grace.spending_money
+    assert_equal 4, @winston.spending_money
+    @lottery.charge_contestants(@mega_millions)
+    expected = { @cash_5 => ["Winston Churchill", "Grace Hopper"],
+    @mega_millions => ["Alexander Aigades", "Frederick Douglas", "Winston Churchill", "Grace Hopper"]}
+    assert_equal expected, @lottery.current_contestants
+    assert_equal 14, @grace.spending_money
+    assert_equal 4, @winston.spending_money
+    assert_equal 5, @alexander.spending_money
+    assert_equal 15, @frederick.spending_money
+    @lottery.charge_contestants(@pick_4)
+    expected = {@cash_5 => ["Winston Churchill", "Grace Hopper"],
+                @mega_millions => ["Alexander Aigades", "Frederick Douglas", "Winston Churchill", "Grace Hopper"],
```

Comment:
Churchill needed to not be ion the `@mega_millions` value, based on the interaction pattern.